Sources/Swiftly/Proxy.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import SwiftlyCore
+
+// This is the allowed list of executables that we will proxy
+let proxyList = [
+    "clang",
+    "lldb",
+    "lldb-dap",
+    "lldb-server",
+    "clang++",
+    "sourcekit-lsp",
+    "clangd",
+    "swift",
+    "docc",
+    "swiftc",
+    "lld",
+    "llvm-ar",
+    "plutil",
+    "repl_swift",
+    "wasm-ld",
+]
```

Comment:
I'm a bit concerned about this--what if a new toolchain has a new executable not on this list? Or what if a toolchain doesn't have a executable on this list? Perhaps being concerned about that is too pessimistic, but it seems somewhat brittle to me. I can't think of any solutions off the top of my head given that we want to support selecting a toolchain via `.swift-version` files, though. The only thing I can think of is requiring users to run `swiftly use` to select the toolchain from the file, but that does seem quite un-ergonomic.

Reply:
The idea of the stable proxy allowed list came out of some necessity, which was to install the proxies at swiftly init time instead of toolchain installation time, but also as a measure of security, which strongly favours allow lists over deny lists.

I can imagine that the commands in this list can be useful to someone in the context of swiftly. We can change this list over time, and manage the proxy symlinks as we do upgrades from version to version. If anyone has suggestions on adding/removing commands from the list I'm interested in hearing your opinions.

This is also one of the reasons that I updated the swiftly design document to clarify that the purpose of swiftly is not to be a fully virtual toolchain. Not all clients of the toolchain will be able to work within the abstraction that swiftly provides. In one case there is a need to reach inside and grab a dylib (.so). There are potentially many others where someone would like to reach in for a particular header file, or other resources. Swiftly isn't equipped to symlink, or proxy, everything in there. It's just not practical.

So, for cases where someone needs access to something more obscure, there is the escape hatch of the `swiftly use --print-location` to get at the real toolchain path where they can reach into the internals as needed at the cost of being optionally dependent on swiftly for that capability.